state/relationunit.go

```diff
@@ -613,34 +613,29 @@ func NetworksForRelation(
 	// is bound to the default space, we need to look up the ingress
 	// address info which is aware of cross model relations.
 	if boundSpace == environs.DefaultSpaceName || err != nil {
-		_, crossmodel, err := rel.RemoteApplication()
```

Comment:
We need the cross model check.
If not cross model, we must not return the public address